custom_components/badnest/api.py

```diff
@@ -362,6 +366,33 @@ def thermostat_set_temperature(self, device_id, temp, temp_high=None):
             self.login()
             self.thermostat_set_temperature(device_id, temp, temp_high)
 
+    def thermostat_set_target_humidity(self, device_id, humidity):
+        if device_id not in self.thermostats:
+            return
+
+        try:
+            self._session.post(
+                f"{self._czfe_url}/v5/put",
+                json={
+                    "objects": [
+                        {
+                            "object_key": f'device.{device_id}',
+                            "op": "MERGE",
+                            "value": {"target_humidity": humidity},
+                        }
+                    ]
+                },
+                headers={"Authorization": f"Basic {self._access_token}"},
+            )
+        except requests.exceptions.RequestException as e:
+            _LOGGER.error(e)
+            _LOGGER.error('Failed to set humidity, trying again')
+            self.thermostat_set_target_humidity(device_id, mode)
```

Comment:
Sorry I didn't catch this earlier.  Looks like a copy-paste error.  Same thing on line 394 below.

```suggestion
            self.thermostat_set_target_humidity(device_id, humidity)
```